tests/core/pyspec/eth2spec/test/electra/epoch_processing/test_apply_pending_deposit.py

```diff
@@ -0,0 +1,117 @@
+from eth2spec.test.context import (
+    spec_state_test,
+    with_electra_and_later,
+)
+from eth2spec.test.helpers.keys import privkeys, pubkeys
+from tests.core.pyspec.eth2spec.test.helpers.deposits import build_deposit_data
+
+@with_electra_and_later
+@spec_state_test
+def test_apply_pending_deposit_add_validator_to_registry(spec, state):
+    amount = 100 
+
+    # choose a value public key that's not in the validator set outside of the mainnet preset of 256
+    index = 2000
+    withdrawal_credentials = spec.BLS_WITHDRAWAL_PREFIX + spec.hash(pubkeys[index])[1:]
+    deposit_data = build_deposit_data(spec,
+                                          pubkeys[index],
+                                          privkeys[index],
+                                          amount,
+                                          withdrawal_credentials,
+                                          signed=True)
+    deposit = spec.PendingDeposit(
+        pubkey=pubkeys[index],
+        withdrawal_credentials= withdrawal_credentials,
+        amount=amount,
+        slot=spec.GENESIS_SLOT,
+        signature=deposit_data.signature,
+        )
+    old_validator_count = len(state.validators)
+    assert spec.apply_pending_deposit(state,deposit) == True
+    # validator count should increase by 1
+    assert len(state.validators) == old_validator_count+1
+
+@with_electra_and_later
+@spec_state_test
+def test_apply_pending_deposit_not_withdrawable_epoch_while_exiting(spec, state):
+    amount = 100 
+    # validator exit epoch must be less than far future
+    state.validators[0].exit_epoch = spec.FAR_FUTURE_EPOCH - 1
+    state.validators[0].withdrawable_epoch = spec.FAR_FUTURE_EPOCH 
+  
+    deposit = spec.PendingDeposit(
+        pubkey=state.validators[0].pubkey,
+        withdrawal_credentials= state.validators[0].withdrawal_credentials,
+        amount=amount,
+        slot=spec.GENESIS_SLOT,
+        )
+    assert spec.apply_pending_deposit(state,deposit) == False
+
+@with_electra_and_later
+@spec_state_test
+def test_apply_pending_deposit_withdrawable_epoch_while_exiting(spec, state):
+    amount = 100 
+    state.slot = spec.SLOTS_PER_EPOCH * 2
+    # validator exit epoch must be less than far future
+    state.validators[0].exit_epoch = spec.FAR_FUTURE_EPOCH - 1
+    state.validators[0].withdrawable_epoch = 0
```

Comment:
It would be good to add a test case where the withdrawable epoch is `get_current_epoch(state)` since the criteria is `<=`  when processing and that would cover the equal case